packages/now-build-utils/src/detect-builders.ts

```diff
@@ -971,10 +971,10 @@ function getRouteResult(
       }
 
       rewriteRoutes.push(...dynamicRoutes);
-      rewriteRoutes.push({
-        src: '^/api(/.*)?$',
+      errorRoutes.push({
         status: 404,
-        continue: true,
+        src: '^/api(/.*)?$',
+        dest: '/_vc_no_match_',
       });
```

Comment:
@dav-is What are your thoughts on having two `handle: error` routes?

Can we handled this correctly in production today and use the default 404 page for `/api` and the custom `/404.html` page for all others?

```js
  [
      {
        status: 404,
        src: '^/api(/.*)?$',
        dest: '/_vc_no_match_',
      },
      {
        status: 404,
        src: '^/(?!.*api).*$',
        dest: '/404.html',
      },
  ]
```

Reply:
What is `/_vc_no_match`?

Reply:
`^/(?!.*api).*$` also matches `/test/api` is that intended?

Reply:
I thought we had to use `dest` so I used something that wouldn't match: `_vc_no_match`

I'll see if it works without the `dest`.

The two `src` matches here are supposed to be opposites: one matches `/api` and paths below, and the other matches everything besides `/api`. So yes, it is intended.

Reply:
`dest` is only required when changing the path. If it's in handle: error. It must not exist

How is `/test/api` and `/api` opposites? It's not in the API directory

Reply:
Oh sorry I meant to say, it doesn't match `/docs/api`, so it wouldn't see the custom 404 page

Reply:
> Oh sorry I meant to say, it doesn't match /docs/api, so it wouldn't see the custom 404 page

Oh yes that is a problem then. Maybe the solution is to use a wildcard to match anything and the order should make it work correctly?

```jsoono
  [
      {
        status: 404,
        src: '^/api(/.*)?$',
        dest: '/_vc_no_match_',
      },
      {
        status: 404,
        src: '^/(.*)*$',
        dest: '/404.html',
      },
  ]
```

Reply:
I think that'd work

Reply:
Remember the `.html` is conditional on `cleanUrls`

Reply:
Why is it `^/(.*)*$` and not `^/.*$`?